mscore/events.cpp

```diff
@@ -715,7 +715,7 @@ static const char* stateName(ViewState s)
             case ViewState::FOTO_DRAG_OBJECT:   p = "FOTO_DRAG_OBJECT"; break;
             case ViewState::FOTO_LASSO:         p = "FOTO_LASSO";       break;
             }
-      return p;
+      return p.c_str();
```

Comment:
... and change this back to
```
return p;
```
?

Reply:
I don't see any reason to use raw strings. It is hard to debug, it needs to be controlled over usage and why? What is wrong with qDebug output you mentioned above?

Reply:
events.cpp:Ms::ScoreView::changeState: changeState s�_�  -> s�_�

Reply:
The function returns a `const char*`. I don't see any reason to construct a `std::string` just to convert it back to a `const char*`.

Reply:
How about that?
```
static const char* stateName(ViewState s)
      {
      switch (s) {
            case ViewState::NORMAL:             return "NORMAL";
            case ViewState::DRAG:               return "DRAG";
            case ViewState::DRAG_OBJECT:        return "DRAG_OBJECT";
            case ViewState::EDIT:               return "EDIT";
            case ViewState::DRAG_EDIT:          return "DRAG_EDIT";
            case ViewState::LASSO:              return "LASSO";
            case ViewState::NOTE_ENTRY:         return "NOTE_ENTRY";
            case ViewState::PLAY:               return "PLAY";
            case ViewState::ENTRY_PLAY:         return "ENTRY_PLAY";
            case ViewState::FOTO:               return "FOTO";
            case ViewState::FOTO_DRAG:          return "FOTO_DRAG";
            case ViewState::FOTO_DRAG_EDIT:     return "FOTO_DRAG_EDIT";
            case ViewState::FOTO_DRAG_OBJECT:   return "FOTO_DRAG_OBJECT";
            case ViewState::FOTO_LASSO:         return "FOTO_LASSO";
            }
      return "";
      }
```

Reply:
Works for me.

Reply:
sure, waiting for the PR :)

Reply:
@lasconic @mattmcclinch done. See https://github.com/musescore/MuseScore/pull/3502.